snap-geotiff/src/main/java/org/esa/snap/dataio/geotiff/TiffTagToMetadataConverter.java

```diff
@@ -115,7 +115,11 @@ private static MetadataAttribute generateMetadataAttribute(TIFFField tiffField)
             if (geoTiffTag.hasValueNames()) {
                 sb.append(geoTiffTag.getValueName(tiffField.getAsInt(i)));
             } else {
-                sb.append(tiffField.getValueAsString(i));
+                try{
+                    sb.append(tiffField.getValueAsString(i));
+                } catch (Exception e){
+                    sb.append("99999");
```

Comment:
Theads provides wrong information to the user. Assuming that the value is 99999 when it could actually be anything.
Again, we have a format issue here and it might be better to throw an exception. If this is wrong, what else might be wrong with the file?
Instead, of "99999" something like "READERROR", OR "UNKNOWN" would be better.